legacy/firmware/fsm_msg_ethereum.h

```diff
@@ -262,12 +262,17 @@ void fsm_msgEthereumSignTypedHash(const EthereumSignTypedHash *msg) {
     layoutHome();
     return;
   }
-  layoutConfirmHash(&bmp_icon_warning, _("EIP-712 message hash"),
-                    msg->message_hash.bytes, 32);
-  if (!protectButton(ButtonRequestType_ButtonRequest_Other, false)) {
-    fsm_sendFailure(FailureType_Failure_ActionCancelled, NULL);
-    layoutHome();
-    return;
+
+  // No message hash when setting primaryType="EIP712Domain"
+  // https://ethereum-magicians.org/t/eip-712-standards-clarification-primarytype-as-domaintype/3286
+  if (msg->message_hash.size) {
```

Comment:
see above, check `msg->has_message_hash` instead

Reply:
Done in https://github.com/trezor/trezor-firmware/pull/2036/commits/2d1a145b172ce6e20cce496e9ba2ba05e4863f76, although it's untested, so I'll CI test it for me.